### PR TITLE
query_range: Validate that start comes before end

### PR DESCRIPTION
**Pull request by amorken**
This should take care of https://github.com/prometheus/prometheus/issues/2140

### CODE REVIEW

File: web/api/v1/api.go [hunk 1]
```diff
@@ -182,6 +182,11 @@ func (api *API) queryRange(r *http.Request) (interface{}, *apiError) {
 	if err != nil {
 		return nil, &apiError{errorBadData, err}
 	}
+	if end <= start {
+		err := errors.New("start timestamp must be before end timestamp. Try switching them around")
```
Comment (beorn7) [inline web/api/v1/api.go]: Both timestamp being equal can be a legal and meaningful query.

The error message itself should not contain periods. I'd just leave out the 2nd half.

**Comment by beorn7** [comment]
This is a bugfix and should be merged into release-1.2 rather than master.
(We don't know if there will be a 1.2.3 but if 1.3 gets delayed for longer, we might have it, or because it looks nice, or because we want K8s users that cannot upgrade to 1.3 right now to receive the bugix too

**Comment by fabxc** [comment]
I wanted to cut 1.3.0 today anyway.

**Comment by beorn7** [comment]
Let's still have this in 1.2 for those that cannot change their K8s SD config easily.

**Comment by amorken** [comment]
OK, I've retargeted the change to release-1.2, and made updates to match @beorn7's comments.

**Comment by beorn7** [comment]
:+1: 
@fabxc as the one being more familiar with the API, does this look good to you, too?

**Comment by fabxc** [comment]
Thanks :+1:

**Pull request merged**